### PR TITLE
CDAP-7014 Update log saver default values for file size and max buckets

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-7014
Build - http://builds.cask.co/browse/CDAP-RUT35-1

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java [hunk 1]
```diff
@@ -96,11 +96,13 @@
     Preconditions.checkArgument(retentionDurationDays > 0,
                                 "Log file retention duration is invalid: %s", retentionDurationDays);
 
-    long maxLogFileSizeBytes = cConf.getLong(LoggingConfiguration.LOG_MAX_FILE_SIZE_BYTES, 20 * 1024 * 1024);
+    // Keep max file size a little less than the default HDFS block size (128 MB)
```
Comment (sagarkapare) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java:99]: would be good to add the reason for this change.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java [hunk 2]
```diff
@@ -96,11 +96,13 @@
...
+    long maxLogFileSizeBytes = cConf.getLong(LoggingConfiguration.LOG_MAX_FILE_SIZE_BYTES, 100 * 1000 * 1000);
     Preconditions.checkArgument(maxLogFileSizeBytes > 0,
                                 "Max log file size is invalid: %s", maxLogFileSizeBytes);
 
-    int syncIntervalBytes = cConf.getInt(LoggingConfiguration.LOG_FILE_SYNC_INTERVAL_BYTES, 50 * 1024);
+    // Sync interval should be around 10 times smaller than file size as we use sync points to navigate
```
Comment (sagarkapare) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java:104]: not sure what it means `we use sync points to navigate`, better to elaborate?

**Comment by sagarkapare** [comment]
@poornachandra minor comments, otherwise LGTM

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java (hunk 1 above)
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java:99]: Even if a file uses space less than the block size, HDFS still allocates space equal to default block size. So it is good to create file sizes a little less than the block size.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java (hunk 2 above)
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaLogWriterPlugin.java:104]: In an avro file, sync points are written at regular intervals. We can jump from sync point to another, and hence can use the sync points to navigate the file when we need to search for something in the file.

**Comment by poornachandra** [comment]
Build is green, merging.

**Pull request merged**